sentry/src/main/java/io/sentry/NoOpSerializer.java

```diff
@@ -28,15 +28,17 @@ public Session deserializeSession(Reader reader) {
   }
 
   @Override
-  public SentryEnvelope deserializeEnvelope(InputStream inputStream) {
+  public Transaction deserializeTransaction(Reader reader) {
     return null;
   }
 
   @Override
-  public void serialize(SentryEvent event, Writer writer) {}
+  public SentryEnvelope deserializeEnvelope(InputStream inputStream) {
+    return null;
+  }
 
   @Override
-  public void serialize(Session session, Writer writer) throws IOException {}
+  public <T> void serialize(T entity, Writer writer) throws IOException {}
```

Comment:
Not propsing to change this but:
What I dislike about this approach is that now this method compiles with any `T` and we surely can't serialize anything. It's a leaky abstraction

Reply:
Technically we can serialize anything that Gson is able to serialize. If we want to make it explicit what can be serialized perhaps we can introduce a a marker interface?

Reply:
yeah a marker interface would be ideal, but we could add this as a task along with moving away from reflection for serializing things, so we do a bigger refactoring and just one breaking change.

Reply:
👍